internal/internal.go

```diff
@@ -461,9 +461,9 @@ func (p *envoyExtAuthzGrpcServer) check(ctx context.Context, req any) (*ext_auth
 		}, stop, nil
 	}
 
-	if ctx.Err() != nil {
-		err = errors.Wrap(ctx.Err(), "check request timed out before query execution")
-		internalErr = newInternalError(CheckRequestTimeoutErr, err)
+	if thrownError, ok := checkCtxError(ctx); ok {
```

Comment:
Go doesn't really have the notion of "thrown errors", errors are values and are not handled as exceptions. Could you name this something like `ctxErr`

Reply:
fixed